pkg/storage/cacher.go

```diff
@@ -405,10 +408,15 @@ func (c *cacheWatcher) add(event watchCacheEvent) {
 }
 
 func (c *cacheWatcher) sendWatchCacheEvent(event watchCacheEvent) {
-	curObjPasses := event.Type != watch.Deleted && c.filter(event.Object)
+	curObjPasses := false
```

Comment:
Before we didn't run filter unless the event was deleted.  Now we always run filter, and only check before.  The former is moderately more performant.